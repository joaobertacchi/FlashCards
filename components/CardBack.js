// @flow
import React from 'react';
import { View, Text, TouchableOpacity } from 'react-native';

type Props = {
  answer: string,
  isCorrect?: boolean,
  onShowFront: () => void,
  onVote: (correct: boolean) => () => void,
};

const CardBack = ({
  answer,
  isCorrect,
  onShowFront,
  onVote,
}: Props) => (
  <View>
    <Text>{answer}</Text>
    <TouchableOpacity onPress={onShowFront}>
      <Text>Show question</Text>
    </TouchableOpacity>
    {isCorrect !== undefined && (
      <Text>{`You marked your answer as ${isCorrect ? 'correct' : 'incorrect'}`}</Text>
    )}
    <TouchableOpacity onPress={onVote(true)}>
      <Text>Correct</Text>
    </TouchableOpacity>
    <TouchableOpacity onPress={onVote(false)}>
      <Text>Incorrect</Text>
    </TouchableOpacity>
  </View>
);

export default CardBack;
