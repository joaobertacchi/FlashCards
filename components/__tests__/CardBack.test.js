import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import NavigationTestUtils from 'react-navigation/NavigationTestUtils';
import CardBack from '../CardBack';

describe('CardBack snapshot', () => {
  jest.useFakeTimers();
  beforeEach(() => {
    NavigationTestUtils.resetInternalState();
  });

  it('renders correctly', () => {
    const tree = renderer
      .create(<CardBack answer="answer" cardNumber={1} cardCount={3} onVote={() => () => null} />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders correctly', () => {
    const tree = renderer
      .create(
        <CardBack
          isCorrect
          answer="answer"
          cardNumber={1}
          cardCount={3}
          onVote={() => () => null}
        />,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders correctly', () => {
    const tree = renderer
      .create(
        <CardBack
          isCorrect={false}
          answer="answer"
          cardNumber={1}
          cardCount={3}
          onVote={() => () => null}
        />,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
