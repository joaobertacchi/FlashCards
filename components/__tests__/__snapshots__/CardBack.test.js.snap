// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardBack snapshot renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "justifyContent": "space-around",
    }
  }
>
  <View
    style={
      Object {
        "flex": 2,
        "justifyContent": "center",
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 24,
          "textAlign": "center",
        }
      }
    >
      answer
    </Text>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "justifyContent": "center",
            "marginBottom": 20,
            "marginHorizontal": 20,
            "marginTop": 10,
          },
        ]
      }
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "white",
            "borderRadius": 7,
            "justifyContent": "center",
            "opacity": 1,
            "padding": 12,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#2f95dc",
                "fontSize": 18,
                "minWidth": 180,
                "textAlign": "center",
              },
              Object {
                "minWidth": undefined,
              },
            ]
          }
        >
          Show question
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "space-around",
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "justifyContent": "center",
            "marginBottom": 20,
            "marginHorizontal": 20,
            "marginTop": 10,
          },
        ]
      }
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "green",
            "borderRadius": 7,
            "justifyContent": "center",
            "opacity": 1,
            "padding": 12,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "white",
                "fontSize": 18,
                "minWidth": 180,
                "textAlign": "center",
              },
              Object {
                "minWidth": 80,
              },
            ]
          }
        >
          Correct
        </Text>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "justifyContent": "center",
            "marginBottom": 20,
            "marginHorizontal": 20,
            "marginTop": 10,
          },
        ]
      }
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "red",
            "borderRadius": 7,
            "justifyContent": "center",
            "opacity": 1,
            "padding": 12,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "white",
                "fontSize": 18,
                "minWidth": 180,
                "textAlign": "center",
              },
              Object {
                "minWidth": 80,
              },
            ]
          }
        >
          Incorrect
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "justifyContent": "center",
          "marginBottom": 20,
          "marginHorizontal": 20,
          "marginTop": 10,
        },
      ]
    }
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "lightgrey",
          "borderColor": "grey",
          "borderRadius": 7,
          "borderWidth": 1,
          "justifyContent": "center",
          "opacity": 1,
          "padding": 12,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "grey",
              "fontSize": 18,
              "minWidth": 180,
              "textAlign": "center",
            },
            Object {
              "minWidth": undefined,
            },
          ]
        }
      >
        Next question
      </Text>
    </View>
  </View>
</View>
`;

exports[`CardBack snapshot renders correctly 2`] = `
<View
  style={
    Object {
      "flex": 1,
      "justifyContent": "space-around",
    }
  }
>
  <View
    style={
      Object {
        "flex": 2,
        "justifyContent": "center",
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 24,
          "textAlign": "center",
        }
      }
    >
      answer
    </Text>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "justifyContent": "center",
            "marginBottom": 20,
            "marginHorizontal": 20,
            "marginTop": 10,
          },
        ]
      }
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "white",
            "borderRadius": 7,
            "justifyContent": "center",
            "opacity": 1,
            "padding": 12,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#2f95dc",
                "fontSize": 18,
                "minWidth": 180,
                "textAlign": "center",
              },
              Object {
                "minWidth": undefined,
              },
            ]
          }
        >
          Show question
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "space-around",
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "justifyContent": "center",
            "marginBottom": 20,
            "marginHorizontal": 20,
            "marginTop": 10,
          },
        ]
      }
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "white",
            "borderColor": "green",
            "borderRadius": 7,
            "borderWidth": 1,
            "justifyContent": "center",
            "opacity": 1,
            "padding": 12,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "green",
                "fontSize": 18,
                "minWidth": 180,
                "textAlign": "center",
              },
              Object {
                "minWidth": 80,
              },
            ]
          }
        >
          Correct
        </Text>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "justifyContent": "center",
            "marginBottom": 20,
            "marginHorizontal": 20,
            "marginTop": 10,
          },
        ]
      }
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "red",
            "borderRadius": 7,
            "justifyContent": "center",
            "opacity": 1,
            "padding": 12,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "white",
                "fontSize": 18,
                "minWidth": 180,
                "textAlign": "center",
              },
              Object {
                "minWidth": 80,
              },
            ]
          }
        >
          Incorrect
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "justifyContent": "center",
          "marginBottom": 20,
          "marginHorizontal": 20,
          "marginTop": 10,
        },
      ]
    }
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#2f95dc",
          "borderRadius": 7,
          "justifyContent": "center",
          "opacity": 1,
          "padding": 12,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "white",
              "fontSize": 18,
              "minWidth": 180,
              "textAlign": "center",
            },
            Object {
              "minWidth": undefined,
            },
          ]
        }
      >
        Next question
      </Text>
    </View>
  </View>
</View>
`;

exports[`CardBack snapshot renders correctly 3`] = `
<View
  style={
    Object {
      "flex": 1,
      "justifyContent": "space-around",
    }
  }
>
  <View
    style={
      Object {
        "flex": 2,
        "justifyContent": "center",
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 24,
          "textAlign": "center",
        }
      }
    >
      answer
    </Text>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "justifyContent": "center",
            "marginBottom": 20,
            "marginHorizontal": 20,
            "marginTop": 10,
          },
        ]
      }
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "white",
            "borderRadius": 7,
            "justifyContent": "center",
            "opacity": 1,
            "padding": 12,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#2f95dc",
                "fontSize": 18,
                "minWidth": 180,
                "textAlign": "center",
              },
              Object {
                "minWidth": undefined,
              },
            ]
          }
        >
          Show question
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "space-around",
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "justifyContent": "center",
            "marginBottom": 20,
            "marginHorizontal": 20,
            "marginTop": 10,
          },
        ]
      }
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "green",
            "borderRadius": 7,
            "justifyContent": "center",
            "opacity": 1,
            "padding": 12,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "white",
                "fontSize": 18,
                "minWidth": 180,
                "textAlign": "center",
              },
              Object {
                "minWidth": 80,
              },
            ]
          }
        >
          Correct
        </Text>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "justifyContent": "center",
            "marginBottom": 20,
            "marginHorizontal": 20,
            "marginTop": 10,
          },
        ]
      }
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "white",
            "borderColor": "red",
            "borderRadius": 7,
            "borderWidth": 1,
            "justifyContent": "center",
            "opacity": 1,
            "padding": 12,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "red",
                "fontSize": 18,
                "minWidth": 180,
                "textAlign": "center",
              },
              Object {
                "minWidth": 80,
              },
            ]
          }
        >
          Incorrect
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "justifyContent": "center",
          "marginBottom": 20,
          "marginHorizontal": 20,
          "marginTop": 10,
        },
      ]
    }
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#2f95dc",
          "borderRadius": 7,
          "justifyContent": "center",
          "opacity": 1,
          "padding": 12,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "white",
              "fontSize": 18,
              "minWidth": 180,
              "textAlign": "center",
            },
            Object {
              "minWidth": undefined,
            },
          ]
        }
      >
        Next question
      </Text>
    </View>
  </View>
</View>
`;
